File: PHP/1_Two_Sum.php
class Solution {
    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($nums, $target) {
        $numIndices = array();
        foreach ($nums as $index => $num) {
            $complement = $target - $num;
            if (isset($numIndices[$complement])) {
                return array($numIndices[$complement], $index);
            }
            $numIndices[$num] = $index;
        }
        return array();
    }
}

File: Ruby/1_Two_Sum.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
    num_indices = {}
    nums.each_with_index do |num, index|
      complement = target - num
      if num_indices.key?(complement)
        return [num_indices[complement], index]
      end
      num_indices[num] = index
    end
    []
end